src/main/java/org/checkerframework/checker/androidresource/AndroidResourceVisitor.java

```diff
@@ -0,0 +1,103 @@
+
+    /**
+     * The @ResourceBottom annotation.
+     */
+    protected final AnnotationMirror RESOURCE_BOTTOM =
+            AnnotationBuilder.fromClass(elements, ResourceBottom.class);
+
+    public AndroidResourceVisitor(BaseTypeChecker checker) {
+        super(checker);
+    }
+
+    /**
+     * Gives a warning whenever operands in a binary assignment have a annotation of type @XXXRes.
+     */
+    @Override
+    public Void visitBinary(BinaryTree node, Void aVoid) {
+
+        ExpressionTree leftOperand = node.getLeftOperand();
+        ExpressionTree rightOperand = node.getRightOperand();
+        AnnotatedTypeMirror leftOperandType = atypeFactory.getAnnotatedType(leftOperand);
+        AnnotatedTypeMirror rightOperandType = atypeFactory.getAnnotatedType(rightOperand);
+
+        Kind kind = node.getKind();
+
+//        System.out.println(node.toString() + ", " + node.getKind() + "\n");
+
+        switch (kind) {
+            case EQUAL_TO:
+                break;
+            case NOT_EQUAL_TO:
+                break;
+            default:
+                if (checkAnnotatedTypeHasResOrContainerAnnotations(leftOperandType) || checkAnnotatedTypeHasResOrContainerAnnotations(rightOperandType)) {
+                    checker.report(Result.warning("binary.operation.not.allowed", kind), node);
+                }
+        }
+
+        return super.visitBinary(node, aVoid);
+    }
+
+    /**
+     * Gives a warning whenever expression/variable in a compound assignment have a annotation of type @XXXRes.
+     */
+    @Override
+    public Void visitCompoundAssignment(CompoundAssignmentTree node, Void p) {
+
+        ExpressionTree expression = node.getExpression();
+        ExpressionTree variable = node.getVariable();
+        AnnotatedTypeMirror expressionType = atypeFactory.getAnnotatedType(expression);
+        AnnotatedTypeMirror variableType = atypeFactory.getAnnotatedType(variable);
+
+        Kind kind = node.getKind();
+
+//        System.out.println(node.toString() + ", " + node.getKind() + "\n");
+
+        if (checkAnnotatedTypeHasResOrContainerAnnotations(expressionType) || checkAnnotatedTypeHasResOrContainerAnnotations(variableType)) {
+            checker.report(Result.warning("compound.assignment.not.allowed", kind), node);
+        }
+
+        return super.visitCompoundAssignment(node, p);
+    }
+
+    /**
+     * Method to check XXXRes or XXXContainer annotations.
+     *
+     * @return whether the given [AnnotatedTypeMirror] contains the XXXRes or XXXContainer annotations or not.
+     */
+    public boolean checkAnnotatedTypeHasResOrContainerAnnotations(AnnotatedTypeMirror annotatedTypeMirror) {
+
+        AnnotationMirror annotationMirror = annotatedTypeMirror.getAnnotationInHierarchy(RESOURCE_TOP);
+
+        if (annotationMirror == null)
+            return false;
+        else if (annotationMirror.equals(RESOURCE_TOP) || annotationMirror.equals(RESOURCE_BOTTOM))

```

Comment:
Don't use `equals` to compare `AnnotationMirrors`. Use `AnnotationUtils.areSame`.
https://github.com/typetools/checker-framework/blob/master/javacutil/src/main/java/org/checkerframework/javacutil/AnnotationUtils.java#L91

Reply:
Fixed.